custom_components/pun_sensor/sensor.py

```diff
@@ -113,88 +112,88 @@ def __init__(self, coordinator: PUNDataUpdateCoordinator, tipo: int) -> None:
         self._native_value = 0
 
     def _handle_coordinator_update(self) -> None:
-        """Gestisce l'aggiornamento dei dati dal coordinator"""
-        self._available = self.coordinator.orari[self.tipo] > 0
-        if (self._available): self._native_value = self.coordinator.pun[self.tipo]
+        """Gestisce l'aggiornamento dei dati dal coordinator."""
+        if len(self.coordinator.pun_data.pun[self.fascia]) > 0:
+            self._available = True
+            self._native_value = self.coordinator.pun_values.value[self.fascia]
```

Comment:
Facendo così, però, non setti mai `self._available` a `False`, nel caso non ci fossero più dati (teoricamente è improbabile, in quanto si recuperano i precedenti, però non è detto al cambio mese... **io non rischierei**).

Reply:
in che senso? in teoria il valore di default del self.available e' false (definito a 111) quindi a meno che quella condizione non e' true il self.available rimane false in teoria

Reply:
Sì, ma credo solo all'inizio quando viene creato il sensore. Poi, nel caso di un aggiornamento del coordinator che non produce dati a sufficienza (quindi `len(...) = 0`) il sensore dovrebbe tornare non disponibile, quindi `False`.
Penso che con questo metodo non accada.

Reply:
Si ci stavo pensando anche io effettivamente, correggo tanto male non fa' di sicuro 👌